src/Tribe/Settings.php

```diff
@@ -607,13 +607,13 @@ public function deleteOptions() {
 		public function get_url( array $args = array() ) {
 			$defaults = array(
 				'page' => $this->adminSlug,
+				'parent' => self::$parent_page,
 			);
 
 			// Allow the link to be "changed" on the fly
 			$args = wp_parse_args( $args, $defaults );
-			$parent = self::$parent_page;
 
-			$url = admin_url( $parent );
+			$url = admin_url( $args['parent'] );
```

Comment:
Wondering just a couple of things here:

1. Should we unset `$args['parent']` after this call to keep the query args of the resulting URL nice and clean?
2. Are we sure there's no problem if a URL generated with a non-default `$args['parent']` is used (given the submenu page is always registered using `self::$parent_page` - line 212)?

For that 2nd point in particular I'm happy if you've checked this out, really just wanted to make sure it's been covered (as it would _seem_ on initial reading that the result would be we'd end up trying to access a submenu page that hasn't actually been registered).